purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -160,16 +160,17 @@ class Purchases internal constructor(
     init {
         identityManager.configure(backingFieldAppUserID)
 
-        dispatch {
-            ProcessLifecycleOwner.get().lifecycle.addObserver(lifecycleHandler)
-        }
-
         billing.stateListener = object : BillingAbstract.StateListener {
             override fun onConnected() {
                 postPendingTransactionsHelper.syncPendingPurchaseQueue(allowSharingPlayStoreAccount)
             }
         }
         billing.purchasesUpdatedListener = getPurchasesUpdatedListener()
+
+        dispatch {
+            ProcessLifecycleOwner.get().lifecycle.addObserver(lifecycleHandler)
+        }
```

Comment:
This is the fix. Moved observing the process lifecycle to happen after the billing listeners have been set. I believe it shouldn't cause any issues as far as I could see in the `onAppForegrounded` callback and my tests didn't reveal any errors either.

Reply:
let's add a comment if the order of operations is important, feels like this could easily get rolled back by accident in a refactor